ai/csharp/detect-openai.yaml

```diff
@@ -0,0 +1,17 @@
+rules:
+  - id: detect-openai
+    languages:
+      - csharp
+    severity: INFO
+    message: "Possibly found usage of AI: OpenAI"
+    pattern-either:
+      - pattern: using OpenAI.$ANYTHING
+      - pattern: ChatClient $CLIENT = ...;
```

Comment:
Could change this to a typed metavariable `(ChatClient $CLIENT)` - will add an additional finding on line 7 of your test file.

Downside - more findings. Upside - findings as a rough proxy for "amount of LLM in code"

Reply:
I struggled to get this to work. [https://semgrep.dev/playground/s/6JZQj](https://semgrep.dev/playground/s/6JZQj?editorMode=advanced) and [https://semgrep.dev/playground/s/oqrpn](https://semgrep.dev/playground/s/oqrpn?editorMode=advanced) both error

Reply:
solved in DM